### PR TITLE
feat: multiselect add toggle multiple support

### PR DESCRIPTION
**Pull request by jatinn**
Took a shot at adding support for support toggling multiple options as mentioned in #180

Some notable changes:
- key bindings displayed are now based on the settings of the field rather than hard-coded
- select-all & limit are mutually exclusive
- ignores filter commands if filterable is false (believe it's a bug but could be a breaking change for folks if they relied on filter commands without setting the `Filterable(true)` on the field)

### CODE REVIEW

File: keymap.go [hunk 1]
```diff
@@ -162,6 +164,8 @@ func NewDefaultKeyMap() *KeyMap {
 			HalfPageDown: key.NewBinding(key.WithKeys("ctrl+d"), key.WithHelp("ctrl+d", "½ page down")),
 			GotoTop:      key.NewBinding(key.WithKeys("home", "g"), key.WithHelp("g/home", "go to start")),
 			GotoBottom:   key.NewBinding(key.WithKeys("end", "G"), key.WithHelp("G/end", "go to end")),
+			SelectAll:    key.NewBinding(key.WithKeys("a", "A"), key.WithHelp("a", "select all")),
+			ClearAll:     key.NewBinding(key.WithKeys("c", "C"), key.WithHelp("c", "clear selections")),
```
Comment (jatinn) [inline keymap.go:168]: open to suggestions on better help text & key bindings

**Comment by jatinn** [comment]
@maaslalani does this feature make sense with your product vision? If so, anything I can help with to move this forward?

**Comment by tdharris** [comment]
This looks great! I'd definitely use this and would like to see it included.

**Comment by tdharris** [comment]
@maaslalani What do you think on this? Sorry for the poke, this one just seems incredibly useful for me.

**Comment by caarlos0** [comment]
hey! 

toggle select all was added to main!

gonna close this!

sorry for the delay, and thanks for the PR anyway!

**Pull request closed without merging**